state/mapstate/migrate.go

```diff
@@ -58,14 +65,40 @@ func (st *mapStateV2) unmarshal(bs []byte) error {
 	return dec.Decode(st)
 }
 
-// No migration possible, v2 is the latest state
 func (st *mapStateV2) next() migrateable {
+	var mst3 mapStateV3
+	mst3.PinMap = make(map[string]api.PinSerial)
+	for k, v := range st.PinMap {
+		mst3.PinMap[k] = api.PinSerial{
+			Cid:                  v.Cid,
+			Name:                 v.Name,
+			Allocations:          v.Allocations,
+			ReplicationFactorMin: v.ReplicationFactor,
+			ReplicationFactorMax: v.ReplicationFactor,
+			ReplicationFactor:    0,
+		}
+	}
+	return &mst3
+}
+
+type mapStateV3 struct {
+	PinMap  map[string]api.PinSerial
+	Version int
+}
+
+func (st *mapStateV3) unmarshal(bs []byte) error {
```

Comment:
This is getting copied enough for us to consider using this as a default implementation for migrateable (if that's possible in golang).

Reply:
I don't think it's possible to have a default implementation for an interface. Probably not a horrible problem either to have copies of that.